### PR TITLE
[stdlib] `write_to(writer, ...)` -> `writer.write(...)`

### PR DESCRIPTION
**Pull request by yinonburgansky**
use `writer.write(...)` instead of `write_to(writer, ...)`

### CODE REVIEW

**Comment by soraros** [comment]
Nice clean up! Hopefully we can also remove `write_to`.

**Comment by gabrieldemarmiesse** [comment]
Nice, this bring `Formatter` closer to a file-like interface :)

**Comment by ConnorGray** [comment]
Thanks for this great cleanup 🧹! :)

There may be internal uses of `write_to()`, but I can just remove those when I merge this PR. Feel free to update this PR to remove `write_to()` as well, and then I'll sync it in 🙂

File: stdlib/src/collections/optional.mojo [hunk 1]
```diff
@@ -285,9 +285,9 @@ struct Optional[T: CollectionElement](
         """
         var output = String()
         var writer = output._unsafe_to_formatter()
-        write_to(writer, "Optional(")
+        writer.write("Optional(")
         self.format_to(writer)
-        write_to(writer, ")")
+        writer.write(")")
```
Comment (ConnorGray) [inline stdlib/src/collections/optional.mojo:290]: **Suggestion** We could condense this down to a single line:

```suggestion
        writer.write("Optional(", self, ")")
```

Comment (yinonburgansky) [reply stdlib/src/collections/optional.mojo:290]: It errors out, we need conditional conformance for it.

**Comment by yinonburgansky** [comment]
@ConnorGray Done, removed `write_to()`.

**Review by JoeLoser — APPROVED** [review]
LGTM!

**Comment by JoeLoser** [comment]
!sync

**Pull request closed without merging**